app/javascript/components/patient/laboratory/Laboratory.js

```diff
@@ -44,29 +36,92 @@ class Laboratory extends React.Component {
+    });
+  };
+
+  handleChange = event => {
+    this.setState({ [event.target.id]: event.target.value });
+  };
+
+  handleDeleteSubmit = () => {
+    let deleteReason = this.state.delete_reason;
+    if (deleteReason === 'Other' && this.state.delete_reason_text) {
+      deleteReason += ', ' + this.state.delete_reason_text;
+    }
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .delete(window.BASE_PATH + '/laboratories/' + this.props.lab.id, {
+          data: {
+            patient_id: this.props.patient.id,
+            delete_reason: deleteReason,
+          },
+        })
+        .then(() => {
+          this.setState({ loading: false, showDeleteModal: false });
+          location.reload(true);
+        })
+        .catch(error => {
+          this.setState({ loading: false, showDeleteModal: false });
+          reportError(error);
+        });
+    });
+  };
+
   render() {
     return (
       <React.Fragment>
         {!this.props.lab.id && (
-          <Button onClick={() => this.setState({ showModal: true, loading: false })}>
+          <Button onClick={() => this.setState({ showEditModal: true, loading: false })}>
             <i className="fas fa-plus fa-fw"></i>
             <span className="ml-2">Add New Lab Result</span>
           </Button>
         )}
         {this.props.lab.id && (
-          <Button variant="link" onClick={() => this.setState({ showModal: true, loading: false })} className="btn btn-link py-0" size="sm">
-            <i className="fas fa-edit"></i> Edit
-          </Button>
+          <Dropdown>
+            <Dropdown.Toggle
+              id={`vaccine-action-button-${this.props.lab.id}`}
+              size="sm"
+              variant="primary"
+              aria-label={`vaccine-action-button-${this.props.lab.id}`}>
+              <i className="fas fa-cogs fw"></i>
+            </Dropdown.Toggle>
+            <Dropdown.Menu>
+              <Dropdown.Item className="px-4 hi" onClick={() => this.setState({ showEditModal: true, loading: false })}>
+                <i className="fas fa-edit fa-fw"></i>
+                <span className="ml-2">Edit</span>
+              </Dropdown.Item>
+              <Dropdown.Item className="px-4 hi" onClick={this.toggleDeleteModal}>
+                <i className="fas fa-trash fa-fw"></i>
+                <span className="ml-2">Delete</span>
+              </Dropdown.Item>
+            </Dropdown.Menu>
+          </Dropdown>
         )}
-        {this.state.showModal && (
+        {this.state.showEditModal && (
           <LaboratoryModal
             lab={this.props.lab}
             submit={this.submit}
-            cancel={() => this.setState({ showModal: false, loading: false })}
-            editMode={!!this.props.lab.id}
+            cancel={() => this.setState({ showEditModal: false, loading: false })}
+            edit_mode={!!this.props.lab.id}
```

Comment:
why this variable name change? I'd leave it camel case

Reply:
This was actually an existing bug i discovered -- it didn't match the expected pop in the LabModal so the LabModal was actually always saying 'Create' even on update. https://github.com/SaraAlert/SaraAlert/blob/c525f63d8f335a8915540a5c4c83b2338c4fdca9/app/javascript/components/patient/laboratory/LaboratoryModal.js#L133

Reply:
ah I missed that - can we keep the camel case and also change the Update to Edit? we pretty much use edit across the board in the front end

Reply:
Vaccine and Close Contact modals right now use 'Update' -- I don't have a strong opinion either way but as far as these types of modals are concerned 'Update' seems to be the more consistent choice

Reply:
My bad, I was thinking about the modal titles.  Added a separate comment for that